front/lib/resources/content_fragment_resource.ts

```diff
@@ -365,13 +367,68 @@ async function renderFromFileId(
   } else {
     const content = await getProcessedFileContent(workspace, fileId);
 
+    const messageWithFullContent = {
+      role: "content_fragment" as const,
+      name: `inject_${contentType}`,
+      content: [
+        {
+          type: "text",
+          text: `<attachment type="${contentType}" title="${title}">\n${content}\n</attachment>`,
+        } as Content,
+      ],
+    };
+
+    if (contentType !== "text/csv") {
+      return new Ok(messageWithFullContent);
+    }
+
+    // For CSV files we have a custom rule:
+    // We render completely the content if the csv is small or a snippet if the csv is big.
+    // A big CSV is a CSV for which tokens counts for more than 10% of the context size of the model.
+    const tokenSize = await tokenCountForText(content, model);
+    const maxTokens = model.contextSize * 0.1;
```

Comment:
a tiny bit radine on the max tokens IMHO, but loosely held opinion

Reply:
I switched to 30%!